src/Raven.Server/Documents/Handlers/QueriesHandler.cs

```diff
@@ -529,8 +529,11 @@ private void TrafficWatchQuery(IndexQueryServerSide indexQuery)
                 Query = query.Query
             };
 
-            var task = Database.Operations.AddOperation(Database, indexName, operationType,
-                onProgress => operation(Database.QueryRunner, options, onProgress, token), operationId, details, token);
+            var task = 
```

Comment:
I think that continuation few lines below will be done on a wrong task, so context will be disposed too early. Ping me when available, we have similar places to this in other parts of the system. I want to understand what is the cause and if it should be treat globally or not.

Reply:
Pretty sure that `Task.Run(Func<Task>)` will return a task that only completes when the inner task task

Reply:
Ok, but it does suggest that AddOperation should do that internally for all executions of that method. Not only here? There are 20 usages of this.

Reply:
I've moved `Task.Run` inside the `AddOperation`. Waiting for CI.